### PR TITLE
Update modules

### PR DESCRIPTION
**Pull request by asp8200**
For reviewers: I got into a bit of nastiness with `snpeff_cache` - especially in `sarek.nf`. So please check that out.

EDIT: Controlfreec-module not updated, since the handling of ploidy won't be part of the upcoming release of Sarek.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/cnvkit/antitarget/main.nf [hunk 1]
```diff
@@ -2,10 +2,10 @@ process CNVKIT_ANTITARGET {
     tag "$meta.id"
     label 'process_low'
 
-    conda "bioconda::cnvkit=0.9.9 bioconda::samtools=1.16.1"
+    conda "bioconda::cnvkit=0.9.10 bioconda::samtools=1.17"
```
Comment (maxulysse) [inline modules/nf-core/cnvkit/antitarget/main.nf:5]: can you update the depencies table in the CHANGELOG related to this tool?

File: modules/nf-core/cnvkit/antitarget/main.nf [hunk 2]
```diff
@@ -2,10 +2,10 @@ process CNVKIT_ANTITARGET {
...
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.9--pyhdfd78af_0':
-        'biocontainers/cnvkit:0.9.9--pyhdfd78af_0' }"
+        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.10--pyhdfd78af_0':
+        'biocontainers/cnvkit:0.9.10--pyhdfd78af_0' }"
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply modules/nf-core/cnvkit/antitarget/main.nf:8]: we had this before.: https://github.com/nf-core/modules/pull/3651

File: modules/nf-core/controlfreec/makegraph/main.nf [hunk 3]
```diff
@@ -8,7 +8,7 @@ process CONTROLFREEC_MAKEGRAPH {
         'biocontainers/control-freec:11.6--h1b792b2_1' }"
 
     input:
-    tuple val(meta), path(ratio), path(baf)
+    tuple val(meta), path(ratio), path(baf), val(ploidy)
```
Comment (FriederikeHanssen) [inline modules/nf-core/controlfreec/makegraph/main.nf:11]: this requires the ploidy param to be passed in

File: modules/nf-core/cnvkit/antitarget/main.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/nf-core/cnvkit/antitarget/main.nf:8]: yeah, something weird is happening with the cnvkit modules there

[the remark this answers is not in the document]
Comment (maxulysse) [reply modules/nf-core/cnvkit/antitarget/main.nf:8]: the lack of consistency is troubling me

File: modules/nf-core/controlfreec/makegraph/main.nf (hunk 3 above)
Comment (FriederikeHanssen) [reply modules/nf-core/controlfreec/makegraph/main.nf:11]: I think I may have actually added, for #980 which turned out to be way more complicated and dropped on the priority list. Could also patch here for now to keep the old behavior and then fix there for good

Comment (asp8200) [reply modules/nf-core/controlfreec/makegraph/main.nf:11]: Should I define `ploidy` like this
```
ploidy = params.cf_ploidy && !params.cf_ploidy.toString().contains(",") ? [params.cf_ploidy] : params.cf_ploidy.split(',')
```
in `subworkflows/local/bam_variant_calling_somatic_all/main.nf` like you did in #980?

If you feel like it, you can also commit the required changes from #980 on "my" branch `update_modules`.

Comment (FriederikeHanssen) [reply modules/nf-core/controlfreec/makegraph/main.nf:11]: 980 is not at all mature and requires a bit more digging into it. not comfortable committing it as is without more testing. I remember that what you show above also had some issue. in any case it is too long too just add

Comment (asp8200) [reply modules/nf-core/controlfreec/makegraph/main.nf:11]: So I see two solutions for now

1. We don't update controlfreec
2. We set the ploidy parameter to `[]` (or something like) in the call to `MAKEGRAPH`

Comment (maxulysse) [reply modules/nf-core/controlfreec/makegraph/main.nf:11]: I vote for `1 don't update`

File: modules/nf-core/cnvkit/antitarget/main.nf (hunk 1 above)
Comment (asp8200) [reply modules/nf-core/cnvkit/antitarget/main.nf:5]: How do you want that written in the dependencies table?  Like this?

```
| Dependency | Old version               | New version              |
| ---------- | ------------------------- | ------------------------ |
| `cnvkit`   | 0.9.9 (`samtools` 1.16.1) | 0.9.10 (`samtools` 1.17) |
| `multiqc`  | 1.14                      | 1.15                     |
| `tiddit`   | 3.3.2                     | 3.6.1                    |
```

File: modules/nf-core/untar/main.nf [hunk 4]
```diff
@@ -2,7 +2,7 @@ process UNTAR {
     tag "$archive"
     label 'process_single'
 
-    conda "conda-forge::sed=4.7 bioconda::grep=3.4 conda-forge::tar=1.34"
+    conda "conda-forge::sed=4.7 conda-forge::grep=3.11 conda-forge::tar=1.34"
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply modules/nf-core/untar/main.nf:5]: grep 3.4 3.11

we haven't reported channel updates

File: CHANGELOG.md [hunk 5]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#1166](https://github.com/nf-core/sarek/pull/1166) - More info about `--tools`
 - [#1174](https://github.com/nf-core/sarek/pull/1174) - Updating multiqc to v1.15
 - [#1179](https://github.com/nf-core/sarek/pull/1179) - Unhide params `trim_fastq`, `umi_read_structure`, and `aligner`
+- [#1180](https://github.com/nf-core/sarek/pull/1180) - Updating the nf-core modules used by Sarek
```
Comment (FriederikeHanssen) [inline CHANGELOG.md:28]:
```suggestion
- [#1180](https://github.com/nf-core/sarek/pull/1180) - Updating the nf-core modules
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -38,10 +39,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
-| Dependency | Old version | New version |
-| ---------- | ----------- | ----------- |
-| `multiqc`  | 1.14        | 1.15        |
-| `tiddit`   | 3.3.2       | 3.6.1       |
+| Dependency               | Old version               | New version              |
+| ------------------------ | ------------------------- | ------------------------ |
+| `cnvkit`                 | 0.9.9 (`samtools` 1.16.1) | 0.9.10 (`samtools` 1.17) |
+| `ensembl-vep`            | 108                       | 110                      |
+| `grep` (in untar-module) | 3.4                       | 3.11                     |
```
Comment (FriederikeHanssen) [inline CHANGELOG.md:46]:
```suggestion
| `grep`                   | 3.4                       | 3.11                     |
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -38,10 +39,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Dependencies
 
-| Dependency | Old version | New version |
-| ---------- | ----------- | ----------- |
-| `multiqc`  | 1.14        | 1.15        |
-| `tiddit`   | 3.3.2       | 3.6.1       |
+| Dependency               | Old version               | New version              |
+| ------------------------ | ------------------------- | ------------------------ |
+| `cnvkit`                 | 0.9.9 (`samtools` 1.16.1) | 0.9.10 (`samtools` 1.17) |
```
Comment (maxulysse) [inline CHANGELOG.md:44]: not sure about specifying samtools, but it does make sense...

**Review by maxulysse — APPROVED** [review]

File: modules.json [hunk 8]
```diff
@@ -303,14 +303,19 @@
...
                         "branch": "master",
-                        "git_sha": "911696ea0b62df80e900ef244d7867d177971f73",
+                        "git_sha": "ebb27711cd5f4de921244bfa81c676504072d31c",
                         "installed_by": ["modules"]
                     },
                     "msisensorpro/msi_somatic": {
                         "branch": "master",
                         "git_sha": "911696ea0b62df80e900ef244d7867d177971f73",
                         "installed_by": ["modules"]
                     },
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply modules.json:313]: good catch

File: modules/nf-core/untar/main.nf (hunk 4 above)
[the remark this answers is not in the document]
Comment (asp8200) [reply modules/nf-core/untar/main.nf:5]: Should the channel update be reported anywhere?

**Comment by asp8200** [comment]
@nf-core-bot fix linting pretty please

**Pull request merged**